library/multipicker/src/main/java/im/vector/lib/multipicker/utils/ImageUtils.kt

```diff
@@ -23,35 +23,51 @@ import android.graphics.ImageDecoder
 import android.net.Uri
 import android.os.Build
 import androidx.exifinterface.media.ExifInterface
+import kotlinx.coroutines.async
+import kotlinx.coroutines.runBlocking
 import timber.log.Timber
 
 object ImageUtils {
 
     fun getBitmap(context: Context, uri: Uri): Bitmap? {
-        return try {
-            if (Build.VERSION.SDK_INT >= Build.VERSION_CODES.P) {
-                ImageDecoder.decodeBitmap(ImageDecoder.createSource(context.contentResolver, uri))
-            } else {
-                context.contentResolver.openInputStream(uri)?.use { inputStream ->
-                    BitmapFactory.decodeStream(inputStream)
+        var bitmap: Bitmap? = null
+        runBlocking {
+            val job = async {
```

Comment:
I do not really understand what will be the diff with the previous version. Since there is a `runBlocking` block, this is still synchronous, no?

Reply:
this might be acting more as workaround by tricking the main thread guard into thinking it's a generic task than a network task :thinking: 

we probably want to mark the functions as `suspend` and then ensure the entire chain :sob:  is also `suspend` and finally called by a coroutine scope of some sort (most likely a view model scope)

Reply:
@ouchadam is right. I´ve just did rerun my tests, it seems it only partially fixes the issue, so I was a bit too rash. (There are still other parts in the chain called on main thread I did not run into the first time I tested)

Reply:
Each error/problem in `contentResolver.openInputStream` (e.g. `FileNotFoundException` due to some network timeout) still crashes the app :/